### PR TITLE
Feature/at-myhuynh/20191018 - Layout Editor

### PR DESCRIPTION
**Pull request by at-myhuynh**
#17

### CODE REVIEW

File: app/src/at-myhuynh/res/layout/activity_edit_profile.xml [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+            app:layout_constraintStart_toStartOf="@id/guidelineLeft"
+            app:layout_constraintTop_toBottomOf="@+id/tvEdit" />
+
+        <EditText
+            android:id="@+id/edtFullName"
+            style="@style/EditTextStyle"
+            android:layout_width="0dp"
+            android:layout_height="wrap_content"
+            android:hint="@string/edittext_text_full_name_hint"
+            android:importantForAutofill="no"
```
Comment (at-dev-support-01) [inline app/src/at-myhuynh/res/layout/activity_edit_profile.xml]: WARNING: [UnusedAttribute] Attribute `importantForAutofill` is only used in API level 26 and higher (current min is 21)
 This check finds attributes set in XML files that were introduced in a version newer than the oldest version targeted by your application (with the `minSdkVersion` attribute).

This is not an error; the application will simply ignore the attribute. However, if the attribute is important to the appearance or functionality of your application, you should consider finding an alternative way to achieve the same result with only available attributes, and then you can optionally create a copy of the layout in a layout-vNN folder which will be used on API NN or higher where you can take advantage of the newer attribute.

Note: This check does not only apply to attributes. For example, some tags can be unused too, such as the new `<tag>` element in layouts introduced in API 21.

File: app/src/at-myhuynh/res/layout/activity_edit_profile.xml [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+            app:layout_constraintTop_toBottomOf="@+id/edtFullName"
+            app:layout_constraintVertical_bias="0.1" />
+
+        <EditText
+            android:id="@+id/edtBio"
+            style="@style/EditTextStyle"
+            android:layout_width="0dp"
+            android:layout_height="wrap_content"
+            android:hint="@string/edittext_hint_bio"
+            android:importantForAutofill="no"
```
Comment (at-dev-support-01) [inline app/src/at-myhuynh/res/layout/activity_edit_profile.xml]: WARNING: [UnusedAttribute] Attribute `importantForAutofill` is only used in API level 26 and higher (current min is 21)
 This check finds attributes set in XML files that were introduced in a version newer than the oldest version targeted by your application (with the `minSdkVersion` attribute).

This is not an error; the application will simply ignore the attribute. However, if the attribute is important to the appearance or functionality of your application, you should consider finding an alternative way to achieve the same result with only available attributes, and then you can optionally create a copy of the layout in a layout-vNN folder which will be used on API NN or higher where you can take advantage of the newer attribute.

Note: This check does not only apply to attributes. For example, some tags can be unused too, such as the new `<tag>` element in layouts introduced in API 21.

File: app/src/at-myhuynh/res/layout/activity_edit_profile.xml [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+            app:layout_constraintStart_toStartOf="@+id/guidelineLeft"
+            app:layout_constraintTop_toBottomOf="@id/tvInfo" />
+
+        <EditText
+            android:id="@+id/edtEmail"
+            style="@style/EditTextStyle"
+            android:layout_width="0dp"
+            android:layout_height="wrap_content"
+            android:hint="@string/edittext_text_email_hint"
+            android:importantForAutofill="no"
```
Comment (at-dev-support-01) [inline app/src/at-myhuynh/res/layout/activity_edit_profile.xml]: WARNING: [UnusedAttribute] Attribute `importantForAutofill` is only used in API level 26 and higher (current min is 21)
 This check finds attributes set in XML files that were introduced in a version newer than the oldest version targeted by your application (with the `minSdkVersion` attribute).

This is not an error; the application will simply ignore the attribute. However, if the attribute is important to the appearance or functionality of your application, you should consider finding an alternative way to achieve the same result with only available attributes, and then you can optionally create a copy of the layout in a layout-vNN folder which will be used on API NN or higher where you can take advantage of the newer attribute.

Note: This check does not only apply to attributes. For example, some tags can be unused too, such as the new `<tag>` element in layouts introduced in API 21.

File: app/src/at-myhuynh/res/layout/activity_edit_profile.xml [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+            app:layout_constraintTop_toBottomOf="@id/edtEmail"
+            app:layout_constraintVertical_bias="0.05" />
+
+        <EditText
+            android:id="@+id/edtPhone"
+            style="@style/EditTextStyle"
+            android:layout_width="0dp"
+            android:layout_height="wrap_content"
+            android:hint="@string/edittext_hint_phone"
+            android:importantForAutofill="no"
```
Comment (at-dev-support-01) [inline app/src/at-myhuynh/res/layout/activity_edit_profile.xml]: WARNING: [UnusedAttribute] Attribute `importantForAutofill` is only used in API level 26 and higher (current min is 21)
 This check finds attributes set in XML files that were introduced in a version newer than the oldest version targeted by your application (with the `minSdkVersion` attribute).

This is not an error; the application will simply ignore the attribute. However, if the attribute is important to the appearance or functionality of your application, you should consider finding an alternative way to achieve the same result with only available attributes, and then you can optionally create a copy of the layout in a layout-vNN folder which will be used on API NN or higher where you can take advantage of the newer attribute.

Note: This check does not only apply to attributes. For example, some tags can be unused too, such as the new `<tag>` element in layouts introduced in API 21.

File: app/src/at-myhuynh/res/layout/activity_login.xml [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+        app:layout_constraintTop_toBottomOf="@id/tvRemind"
+        app:layout_constraintVertical_bias="0.09" />
+
+    <EditText
+        android:id="@+id/edtEmail"
+        android:layout_width="0dp"
+        android:layout_height="wrap_content"
+        style="@style/EditTextStyle"
+        android:hint="@string/edittext_text_email_hint"
+        android:importantForAutofill="no"
```
Comment (at-dev-support-01) [inline app/src/at-myhuynh/res/layout/activity_login.xml]: WARNING: [UnusedAttribute] Attribute `importantForAutofill` is only used in API level 26 and higher (current min is 21)
 This check finds attributes set in XML files that were introduced in a version newer than the oldest version targeted by your application (with the `minSdkVersion` attribute).

This is not an error; the application will simply ignore the attribute. However, if the attribute is important to the appearance or functionality of your application, you should consider finding an alternative way to achieve the same result with only available attributes, and then you can optionally create a copy of the layout in a layout-vNN folder which will be used on API NN or higher where you can take advantage of the newer attribute.

Note: This check does not only apply to attributes. For example, some tags can be unused too, such as the new `<tag>` element in layouts introduced in API 21.

File: app/src/at-myhuynh/res/layout/activity_login.xml [hunk 6]
```diff
@@ -0,0 +1,183 @@
...
+        android:id="@+id/edtPassword"
+        android:layout_width="0dp"
+        android:layout_height="wrap_content"
+        android:inputType="textPassword"
+        style="@style/EditTextStyle"
+        app:layout_constraintEnd_toEndOf="@+id/guidelineRight"
+        app:layout_constraintStart_toStartOf="@+id/guidelineLeft"
+        app:layout_constraintTop_toBottomOf="@+id/tvPassword"
+        tools:ignore="LabelFor"
+        android:importantForAutofill="no" />
```
Comment (at-dev-support-01) [inline app/src/at-myhuynh/res/layout/activity_login.xml]: WARNING: [UnusedAttribute] Attribute `importantForAutofill` is only used in API level 26 and higher (current min is 21)
 This check finds attributes set in XML files that were introduced in a version newer than the oldest version targeted by your application (with the `minSdkVersion` attribute).

This is not an error; the application will simply ignore the attribute. However, if the attribute is important to the appearance or functionality of your application, you should consider finding an alternative way to achieve the same result with only available attributes, and then you can optionally create a copy of the layout in a layout-vNN folder which will be used on API NN or higher where you can take advantage of the newer attribute.

Note: This check does not only apply to attributes. For example, some tags can be unused too, such as the new `<tag>` element in layouts introduced in API 21.

File: app/src/at-myhuynh/res/layout/activity_sign_up.xml [hunk 7]
```diff
@@ -0,0 +1,206 @@
...
+            app:layout_constraintTop_toBottomOf="@id/tvRemind"
+            app:layout_constraintVertical_bias="0.07" />
+
+        <EditText
+            android:id="@+id/edtFullName"
+            android:layout_width="0dp"
+            android:layout_height="wrap_content"
+            style="@style/EditTextStyleNunito"
+            android:hint="@string/edittext_text_full_name_hint"
+            android:importantForAutofill="no"
```
Comment (at-dev-support-01) [inline app/src/at-myhuynh/res/layout/activity_sign_up.xml]: WARNING: [UnusedAttribute] Attribute `importantForAutofill` is only used in API level 26 and higher (current min is 21)
 This check finds attributes set in XML files that were introduced in a version newer than the oldest version targeted by your application (with the `minSdkVersion` attribute).

This is not an error; the application will simply ignore the attribute. However, if the attribute is important to the appearance or functionality of your application, you should consider finding an alternative way to achieve the same result with only available attributes, and then you can optionally create a copy of the layout in a layout-vNN folder which will be used on API NN or higher where you can take advantage of the newer attribute.

Note: This check does not only apply to attributes. For example, some tags can be unused too, such as the new `<tag>` element in layouts introduced in API 21.

File: app/src/at-myhuynh/res/layout/activity_sign_up.xml [hunk 8]
```diff
@@ -0,0 +1,206 @@
...
+            app:layout_constraintBottom_toBottomOf="parent"
+            app:layout_constraintStart_toStartOf="@id/guidelineLeft"
+            app:layout_constraintTop_toBottomOf="@+id/edtFullName"
+            app:layout_constraintVertical_bias="0.07" />
+
+        <EditText
+            android:id="@+id/edtEmail"
+            android:layout_width="0dp"
+            android:layout_height="wrap_content"
+            android:importantForAutofill="no"
```
Comment (at-dev-support-01) [inline app/src/at-myhuynh/res/layout/activity_sign_up.xml]: WARNING: [UnusedAttribute] Attribute `importantForAutofill` is only used in API level 26 and higher (current min is 21)
 This check finds attributes set in XML files that were introduced in a version newer than the oldest version targeted by your application (with the `minSdkVersion` attribute).

This is not an error; the application will simply ignore the attribute. However, if the attribute is important to the appearance or functionality of your application, you should consider finding an alternative way to achieve the same result with only available attributes, and then you can optionally create a copy of the layout in a layout-vNN folder which will be used on API NN or higher where you can take advantage of the newer attribute.

Note: This check does not only apply to attributes. For example, some tags can be unused too, such as the new `<tag>` element in layouts introduced in API 21.

File: app/src/at-myhuynh/res/layout/activity_sign_up.xml [hunk 9]
```diff
@@ -0,0 +1,206 @@
...
+        <EditText
+            android:id="@+id/edtPassword"
+            android:layout_width="0dp"
+            android:layout_height="wrap_content"
+            app:layout_constraintTop_toBottomOf="@+id/tvPassword"
+            app:layout_constraintStart_toStartOf="@+id/guidelineLeft"
+            app:layout_constraintBottom_toBottomOf="parent"
+            app:layout_constraintEnd_toEndOf="@+id/guidelineRight"
+            app:layout_constraintVertical_bias="0.0"
+            android:importantForAutofill="no"
```
Comment (at-dev-support-01) [inline app/src/at-myhuynh/res/layout/activity_sign_up.xml]: WARNING: [UnusedAttribute] Attribute `importantForAutofill` is only used in API level 26 and higher (current min is 21)
 This check finds attributes set in XML files that were introduced in a version newer than the oldest version targeted by your application (with the `minSdkVersion` attribute).

This is not an error; the application will simply ignore the attribute. However, if the attribute is important to the appearance or functionality of your application, you should consider finding an alternative way to achieve the same result with only available attributes, and then you can optionally create a copy of the layout in a layout-vNN folder which will be used on API NN or higher where you can take advantage of the newer attribute.

Note: This check does not only apply to attributes. For example, some tags can be unused too, such as the new `<tag>` element in layouts introduced in API 21.

File: app/src/at-myhuynh/AndroidManifest.xml [hunk 10]
```diff
@@ -4,12 +4,18 @@
...
         android:allowBackup="true"
+        android:hardwareAccelerated="false"
         android:icon="@mipmap/ic_launcher"
         android:label="@string/app_name"
+        android:largeHeap="true"
         android:roundIcon="@mipmap/ic_launcher_round"
         android:supportsRtl="true"
         android:theme="@style/AppTheme">
-        <activity android:name=".MainActivity">
+        <activity android:name=".MainActivity"></activity>
```
Comment (at-tainguyen) [inline app/src/at-myhuynh/AndroidManifest.xml]: Remove close tag if don't have any child tag

File: app/src/at-myhuynh/res/drawable/ic_around_button.xml [hunk 11]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="utf-8"?>
+<selector xmlns:android="http://schemas.android.com/apk/res/android">
+    <item>
+        <shape android:shape="rectangle">
+            <solid android:color="@color/colorGreen" />
+            <corners android:radius="10dp" />
+        </shape>
+    </item>
+</selector>
```
Comment (at-tainguyen) [inline app/src/at-myhuynh/res/drawable/ic_around_button.xml]: Pls don't forget to add the break line at end of file

File: app/src/at-myhuynh/res/values/dimen.xml [hunk 12]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <dimen name="text_size_15">15sp</dimen>
+    <dimen name="text_size_17">17sp</dimen>
+    <dimen name="text_size_20">20sp</dimen>
+</resources>
```
Comment (at-tainguyen) [inline app/src/at-myhuynh/res/values/dimen.xml]: Pls don't forget to add the break line at end of file

**Pull request merged**